compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -493,28 +493,31 @@ macro_rules! define_queries {
                 &tcx.query_caches.$name
             }
 
+            fn execute_query(tcx: TyCtxt<'tcx>, key: Self::Key) -> Self::Stored {
+                tcx.$name(key)
+            }
+
             #[inline]
-            fn make_vtable(tcx: QueryCtxt<'tcx>, key: &Self::Key) ->
-                QueryVTable<QueryCtxt<'tcx>, Self::Key, Self::Value>
-            {
-                let compute = get_provider!([$($modifiers)*][tcx, $name, key]);
-                let cache_on_disk = Self::cache_on_disk(tcx.tcx, key);
-                QueryVTable {
-                    anon: is_anon!([$($modifiers)*]),
-                    eval_always: is_eval_always!([$($modifiers)*]),
-                    depth_limit: depth_limit!([$($modifiers)*]),
-                    feedable: feedable!([$($modifiers)*]),
-                    dep_kind: dep_graph::DepKind::$name,
-                    hash_result: hash_result!([$($modifiers)*]),
-                    handle_cycle_error: handle_cycle_error!([$($modifiers)*]),
-                    compute,
-                    try_load_from_disk: if cache_on_disk { should_ever_cache_on_disk!([$($modifiers)*]) } else { None },
-                }
+            fn compute(qcx: QueryCtxt<'tcx>, key: Self::Key) -> Self::Value {
```

Comment:
The `inline` here is not very useful, since this method is coerced to a function pointer most of the times. This means that `with_task` gets a fn ptr which calls a fn ptr.
Returning the provider as `fn(TyCtxt<'tcx>, Self::Key) -> Self::Value` may be more interesting: we wan inline this function, and `with_task` directly has a fn ptr.

Reply:
That sounds good, I'll try that.